packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -465,6 +467,30 @@ export class StripeWebhookHandler extends StripeHandler {
       if (this.paypalHelper) {
         await this.paypalHelper.conditionallyRemoveBillingAgreement(customer);
       }
+
+      const plan = event.data.object as Stripe.Plan;
+      const product = await this.stripeHelper.fetchProductById(
+        plan.product as string
+      );
+      const cancellationDetails: Stripe.Subscription.CancellationDetails | null =
+        sub.cancellation_details;
+      const cancellationReason = cancellationDetails?.reason;
+      const voluntaryCancellation: boolean =
+        cancellationReason === 'cancellation_requested';
+
+      await request.emitMetricsEvent('customer.subscription.deleted', {
+        // country_code_source: '',
+        // event_type: '',
+        language: request.app.locale,
+        time: Date.now(),
+        user_id: account?.uid,
+        payment_provider: this.stripeHelper.getPaymentProvider(customer),
+        plan_id: plan.id,
+        product_id: product?.id,
+        provider_event_id: event.id,
+        subscription_id: sub.id,
+        voluntary_cancellation: voluntaryCancellation,
+      });
```

Comment:
When testing this locally, we do see the event in the auth (`error`) log output, however it's missing a number of fields, despite them being passed into this method. Next step is to figure out why.

- Global
	- `country_code_source`
		- Keep this property wherever the existing `source_country` (prior name) property is (see current `fxa_pay_setup - success` events).
	- `user_id`
		- Keep where it normally is for other events (looks to be neither in `event_properties` nor `user_properties`... -_-)
- `event_properties`
	- `payment_provider`
	- `product_id`
	- `provider_event_id`
	- `subscription_id`
	- `voluntary_cancellation`